src/scheduler/resv_info.c

```diff
@@ -1140,6 +1140,9 @@ check_new_reservations(status *policy, int pbs_sd, resource_resv **resvs, server
 			/* Clean up simulated server info */
 			free_server(nsinfo, 1);
 		}
+		/* Something went wrong with reservation confirmation, retry later */
+		if (pbsrc == RESV_CONFIRM_RETRY)
+			return -1;
```

Comment:
Why is this out side of the `if (pbsrc == XXX) else if` block?

Reply:
It's so I don't have to free up everything multiple times.  If I made it part of the if/elseif block, I'd have to free everything and return -1 else free everything and continue on with the for loop.  This way I free everything once and then if pbsrc == RESV_CONFIRM_RETRY, I return -1.